### PR TITLE
Add copy button to Markdown codeblocks

### PR DESCRIPTION
**Pull request by philippjfr**
Implements https://github.com/holoviz/panel/issues/6209

copy_button

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
Would be so great for chat and many other use cases.

**Comment by ahuang11** [comment]
One feature that I really like about ChatGPT is how the copy icon follows the scroll. Is it easy to implement that feature?

https://github.com/user-attachments/assets/86d6096a-2572-447f-b567-a5409a5d292b

File: panel/styles/models/html.less [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.copybtn {
+  position: absolute;
```
Comment (MarcSkovMadsen) [inline panel/styles/models/html.less:2]: I would expect the copy button to have a color aligned with the (dark) theme. But it has not.

https://github.com/user-attachments/assets/f5de2143-3aa5-4e2a-b159-228d33c97529

```python
import panel as pn
from panel.chat import ChatInterface

pn.extension("perspective")

code = """
Some code

```python
a=1
b=a+1
print(a, b)
```
"""

pn.template.FastListTemplate(
    title="Copy Button",
    sidebar=[code],
    main=[code],
).servable()
```

**Comment by Coderambling** [comment]
> One feature that I really like about ChatGPT is how the "Copy Code" follows the scroll position, up to the end of the code, so users don't have to scroll all the way up to copy. Is it easy to implement that feature?
> 
>  Screen.Recording.2024-10-30.at.9.24.55.AM.mov

If the "follow" functionality is tricky to implement, maybe simply put the copy button at the end of the code block, as well as the beginning? Would also be a little less busy than a "following" button.

File: panel/styles/models/html.less (hunk 1 above)
Comment (MarcSkovMadsen) [reply panel/styles/models/html.less]: There could be inspiration here:

- https://docs.streamlit.io/develop/api-reference/text/st.code

Comment (MarcSkovMadsen) [reply panel/styles/models/html.less]: This is how chatgpt looks for me. Could also be inspiration.

https://github.com/user-attachments/assets/3df18770-5a9a-4300-89f5-03e02771c3b7

**Comment by philippjfr** [comment]
Will skip the following behavior for now. Adding two buttons is definitely too busy.

**Pull request merged**